athina/evals/function/functions.py

```diff
@@ -43,7 +73,155 @@ def contains_any(keywords, response=None, case_sensitive=False):
+
+    Args:
+        keywords (str or List[str]): The keyword(s) to search for in the response.
+        response (str): The response string to search within.
+        case_sensitive (bool, optional): If True, the comparison is case-sensitive. Defaults to False.
+
+    Returns:
+        dict: A dictionary containing the result of the check and the reason for the result.
+    """
+    if not case_sensitive:
+        response = response.lower()
+        keywords = list(map(lambda k: k.lower(), keywords))
+
+    found_keywords = []
+    for keyword in keywords:
+        if keyword in response:
+            found_keywords.append(keyword)
+
+    if found_keywords:
+        result = False
+        reason = f"One or more keywords were found in output: " + ", ".join(
+            found_keywords
+        )
+    else:
+        result = True
+        reason = "No keywords found in output"
+
+    return {"result": result, "reason": reason}
+
+def contains_json(response):
+    """
+    Check if the response contains valid JSON.
+
+    Args:
+        response (str): The response string to check for valid JSON.
+
+    Returns:
+        dict: A dictionary containing the result of the JSON check and the reason for the result.
+    """
+    trimmed_output = response.strip()
+    pattern = r'\{(?:\s*"(?:\\.|[^"\\])*"\s*:\s*(?:"(?:\\.|[^"\\])*"|[^{}\[\]:,]+)|[^{}]+)*\}'
+    matches = re.findall(pattern, trimmed_output)
+
+    if matches:
+        results = []
+        errors = []
+        for potential_json_string in matches:
+            try:
+                parsed_json = json.loads(potential_json_string)
+                results.append({"json": parsed_json, "valid": True})
+            except json.JSONDecodeError as e:
+                errors.append({"json": potential_json_string, "valid": False, "error": str(e)})
+        if errors:
+            return {"result": False, "reason": "Output contains a potential JSON but it is invalid", "matches": results, "errors": errors}
+        else:   
+            return {"result": True, "reason": "Output contains JSON", "matches": results}
+    else:
+        return {"result": False, "reason": "Output does not contain JSON"}
+    
+def contains_email(response):
+    """
+    Check if the response contains an email address.
+
+    Args:
+        response (str): The response string to check for an email address.
+
+    Returns:
+        dict: A dictionary containing the result of the email address check and the reason for the result.
+    """
+    return regex(r'[a-zA-Z0-9_.+-]+@[a-zA-Z0-9-]+\.[a-zA-Z0-9-.]+', response)
+
+def contains_phone_number(response):
+    """
+    Check if the response contains a phone number.
+
+    Args:
+        response (str): The response string to check for a phone number.
+
+    Returns:
+        dict: A dictionary containing the result of the phone number check and the reason for the result.
+    """
+    return regex(r'\+?1?\d{9,15}', response)
```

Comment:
Iirc from my testing in the legacy SDK, this regex is not super reliable. 

Also, it will only capture US phone numbers. And it wouldn't capture phone numbers like this: (xxx) xxx-xxxx, which is a very common formatting scheme for US phone numbers.

Maybe we should just remove the eval altogether for now.